UX DoD Requirements (#1490)
kyma-project/console: CONTRIBUTING.md

# Overview

To contribute to this project, follow the rules from the general [CONTRIBUTING.md](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md) document in the `community` repository.

## UX DoD Requirements

### General information

UI developed within the Console and Compass UI should be compliant with [Fiori Design Guidelines](https://experience.sap.com/fiori-design-web/).

### Additional guidelines for modals

* User should be able to close the modal (without saving changes, if any) by clicking anywhere outside of it.
* Modal should support keyboard control:

>     * User should be always able to navigate the modal by the means of keyboard only. While the modal is opened, focus should not leave the modal window.

Reviewer: ```suggestion
    * User should always be able to navigate the modal using only a keyboard. While the modal is opened, focus should not leave the modal window.
```

>     * If modal contains input fields, first of them should be automatically focused on opening the modal. In some cases, we can focus the most important element insted (e.g. invalid one).

Reviewer: ```suggestion
    * If modal contains input fields, the first of them should be automatically focused on opening the modal. In some cases, we can focus on the most important element instead.
```

>     * If a modal contains input fields, the first of them should be automatically focused on opening the modal. In some cases, we can set focus on the most important element instead (e.g. incorrect one).

Reviewer: ```suggestion
    * If the modal contains input fields, the first of them should be automatically focused on opening the modal. In some cases, we can set focus on the most important element instead (e.g. the incorrect one).
```

>     * Main action (e.g. _Confirm_, _Save_, _Apply_) should use _emphasized_ style (read more [here](https://experience.sap.com/fiori-design-web/button)).

Reviewer: ```suggestion
    * Main actions, such as **Confirm**,  **Save**, **Apply**, should use the [emphasized style](https://experience.sap.com/fiori-design-web/button).
```

> * If content of the modal is dynamic, try to avoid resizing the modal itself. What's more, modals' widths should be consistent across the application.

Reviewer: ```suggestion
* If the content of the modal is dynamic, avoid resizing the modal. Modals width should be consistent across the application.
```

> * Remember about [ARIA](https://www.w3.org/WAI/standards-guidelines/aria/) attributes.

Reviewer: ```suggestion
* Follow the [ARIA](https://www.w3.org/WAI/standards-guidelines/aria/) attributes.
```

### Additional guidelines for Compass

> * While creating validable forms, make sure to meet the requirements outlined [here](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-input-validation.md).

Reviewer: ```suggestion
* When creating validable forms, make sure to meet the requirements outlined [here](https://github.com/kyma-incubator/compass/blob/master/docs/compass/03-input-validation.md).
```

> * While using HTML input `type` attribute for validation, make sure to verify if their validation corresponds to the one used on the backend.

Reviewer: ```suggestion
* When using HTML input `type` attribute for validation, make sure to verify if the validation corresponds to the one used on the backend.
```

> * While using HTML input `type` attribute for validation, make sure to verify that their validation corresponds to one used on backend.

Reviewer: ```suggestion
* While using HTML input `type` attribute for validation, make sure to verify if their validation corresponds to the one used on the backend.
```
